### ISSUE TITLE
G-Sync/FreeSync support

### ISSUE THREAD

**Issue by ROMaster2** [mgba-emu/mgba#478]
You can put this under the "won't even touch until someone actually wants this" category.

[This video](https://www.youtube.com/watch?v=2CeZ0xbtfDo) highlights why G-Sync and FreeSync are important for emulation and old games. Since the GBA doesn't use a standard PC framerate, it could be more accurately displayed with those. According the video MAME has some support for them, so it might be available in [it's source code](https://github.com/mamedev/mame) (I can't find it though).

**Comment by endrift** [comment]
I don't have the hardware required to do this myself, but it would be a nice thing to have at some point.

**Comment by rraab8** [comment]
Should Variable Refresh Rate	from HDMI 2.1 spec be added to this?

**Comment by ROMaster2** [comment]
It's been a little over five years and the hardware is more commonplace. Are you able to do it now?

**Comment by endrift** [comment]
I do in fact have the hardware for it now. I should schedule this for a release, especially if it's actually doable within Qt (I don't know; I made need some advice from @ahigerd on this).

**Comment by Immersion95** [comment]
It works perfectly here, this should be closed.

**Comment by ahigerd** [comment]
Just because it works doesn't mean it ACTUALLY works. There has been no work done on this task yet and there's still actual effort necessary to get proper VRR support implemented.

**Comment by Immersion95** [comment]
> It works perfectly here, this should be closed.
> 
> Use Audio sync which use the real fps of the GBA and disable Video sync.

It didn't work perfectly, Audio sync created video stutter and video sync created audio crackle.

This commit fixes video sync perfectly https://github.com/mgba-emu/mgba/commit/f51cb153d109f842fdc5325715be97a71fcacd28

Just use the dev build and enable video sync+disable audio sync. It is still not implemented in the latest release.

**Comment by adamhigerd** [comment]
@Immersion95 That commit does not fix it perfectly. It will work better for some people but the underlying problem still exists.

**Comment by adamhigerd** [comment]
@Immersion95 Yes, but that's just you. It depends on the details of your hardware. I was just supporting someone on the latest dev build yesterday that was hearing pretty notable stutter. This is not a general fix and should not be presented as such.
